ModularTegustation/tegu_items/gadgets/unpowered.dm

```diff
@@ -295,3 +295,69 @@
 	if(istype(H) && (H?.mind?.assigned_role == "Clerk"))
 		return TRUE
 	return FALSE
+
+
+
+
+//Tool E.G.O extractor
+/obj/item/tool_extractor
+	name = "Enkephalin Resonance Unit"
+	desc = "A specialized set of tools that allows E.G.O extraction from tool abnormalities."
+	icon = 'icons/obj/storage.dmi'
+	icon_state = "RPED"
+	w_class = WEIGHT_CLASS_BULKY
+	slot_flags = ITEM_SLOT_BELT
+	var/stored_enkephalin = 0
+	var/maximum_enkephalin = 250
+	var/ego_selection
+	var/ego_array
+
+/obj/item/tool_extractor/examine(mob/user)
+	. = ..()
+	. += "Currently storing [stored_enkephalin]/[maximum_enkephalin] enkephalin."
+
+/obj/item/tool_extractor/attack_obj(obj/O, mob/living/carbon/user)
+	if(user.a_intent == INTENT_HARM)
+		return ..()
+	if(istype(O ,/obj/machinery/computer/extraction_cargo))//console stuff here
+		var/drawn_amount = 50
```

Comment:
Can you make this a variable glued to the tool itself and allow it to be adjusted by using it in hand like an EMAIS where it switches dose amount? For example, being able to switch between 25 and 50 drain.